lib/rules/no-this-translate.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @fileoverview Disallow the use of this.translate
+ * @author Automattic
+ * @copyright 2016 Automattic. All rights reserved.
+ * See LICENSE.md file in root directory for full license.
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const ERROR_MESSAGE = 'Use localize( ReactComponent ) instead of this.translate';
+
+module.exports = {
+	meta: {
+		docs: {
+			description: 'Disallow the use of this.translate',
+			category: 'Deprecation',
```

Comment:
I guess this is up to us to name, but it appears ESLint's core rules have adopted "Deprecated", not "Deprecation". Should we use this for consistency?

http://eslint.org/docs/rules/#deprecated